src/Model/ArticleManager.php

```diff
@@ -25,14 +25,33 @@ public function __construct(\PDO $pdo)
     /*
     *searching article by category and by name(when searching by the client
     */
-      public function searchArticle(string $category,string $search=''): array
-      {
-          $searching = '';
+     public function searchArticle(string $category,string $search=''): array
+      {   $searching = '';
           if (!empty($search)) {
-              $searching = "AND name LIKE '%$search%'";
+              $searching = "AND name LIKE :search";
+          }
+          $statement=$this->pdo->prepare('SELECT * FROM ' . $this->table . " WHERE   category =:category $searching");
+          $statement->setFetchMode(\PDO::FETCH_CLASS, $this->className);
+          $statement->bindValue('category', $category, \PDO::PARAM_STR);
+          if (!empty($search)) {
+              $statement->bindValue('search', "%$search%", \PDO::PARAM_STR);
+          }
+          if ($statement->execute()) {
+              return $statement->fetchAll();
           }
-          return $this->pdo->query('SELECT * FROM ' . $this->table . " WHERE   category ='$category' $searching", \PDO::FETCH_CLASS, $this->className)->fetchAll();
       }
+    /*
+ *searching article by name
+ */
+    public function searchArticleGeneral(string $search=''): array
```

Comment:
pour refactoriser, tu pourrais n'utiliser que searchArticle()
dans le cas d'une recherche globale, tu appellerais ta méthode avec searchArticle(null, $maRecherche) (null pour $category).
Et dans la méthode, tu fais une condition (comme pour $search) qui affiche le bout de requete sur la category (et le bindValue optionnel également). Il faudra juste bien gérer le 'AND' entre les deux.

On en parle si tu veux, mais ça serait plus DRY et intéressant en terme de code ;-)